test/test_samplechanger_routes.py

```diff
@@ -0,0 +1,86 @@
+import json
+import random
+
+# Python 2 and 3 compatibility
+try:
+    unicode
+except:
+    unicode = str
+
+from fixture import client
+
+
+def test_get_sample_list(client):
+    """
+    Checks retrieval of the samples list from lims
+    """
+    resp = client.get("/mxcube/api/v0.1/sample_changer/samples_list")
+    data = json.loads(resp.data)
+
+    assert isinstance(data["sampleList"], dict)
+    assert isinstance(data["sampleOrder"], list)
+
+def test_get_sc_state(client):
+    """
+    Checks retrieval of the sample changer state
+    """
+    resp = client.get("/mxcube/api/v0.1/sample_changer/state")
+    data = json.loads(resp.data)
+
+    assert isinstance(data["state"], unicode)
+
+def test_get_loaded_sample(client):
+    """
+    Checks retrieval of the sample changer loaded sample
+    """
+    resp = client.get("/mxcube/api/v0.1/sample_changer/loaded_sample")
+    data = json.loads(resp.data)
+
+    assert isinstance(data["address"], unicode)
+    assert isinstance(data["barcode"], unicode)
+
+def test_get_sc_contents_view(client):
+    """
+    Checks retrieval of the sample changer contents
+    """
+    resp = client.get("/mxcube/api/v0.1/sample_changer/contents")
+    data = json.loads(resp.data)
+    print(data)
+    assert isinstance(data["children"], list)
+    assert len(data["children"]) == 17  # pucks
```

Comment:
Be careful here, because this only tests this specific implementation of the sample changer in this case SampleChanger Mockup. Other sample changers might have other layouts. It would probably, if possible, be better to get the actual number of baskets and samples from the sample changer HardwareObject.

Reply:
But we run all tests with mockups

Reply:
Sure, but we might want to use a different layout one day or even test against a real sample changer. It would be good if we could avoid hard coding values like these. There is also a print left in the test, is that intentional ?

Reply:
modify a bit the test and added a new url for getting the capacity